### DISCUSSION TITLE
Where to take this project ...

### DISCUSSION THREAD

**Discussion by swade1987** [swade1987/deprek8ion#32] [Q&A]
## 👋 Welcome!
  I'm using Discussions as a place to connect with other members of our community. I hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other project users.
  * Welcome others and are open-minded. Remember that this is a product we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

## 👋 Feedback!

I am looking for where to take this project, would it be more useful to have: 

1. Tagged images for specific minor versions that deny any deprecated APIs and warn on up and coming ones. 

2. Stick with the way it is currently?

Looking forward to your feedback ...

**Comment by valorl** [comment]
From my limited usage so far, the use-case is along the lines of "I want to be ready for 1.22. Where do I need to change things?". 

From that perspective I would want something like 1️⃣ except I guess it should perhaps deny _removed_ ones and warn on deprecated ones? Or maybe it could actually say what's been deprecated, what's been removed, etc. I understand that that would require some more code around conftest/OPA resultsets, maybe running conftest multiple times / with different policy sets for deprecated vs removed and so on.


So the point would be to pick a K8S version, and see which of my current manifests would be problematic if I were to upgrade to that version. I feel like that can probably be achieved/organized with different approaches. Having git and image tags represent the "target" K8S version sounds good enough. Although I wonder if that would hinder the versioning and maintenance of deprek8ion itself.

Thoughts 🙂 ? Does this make sense ? Are there any other perspectives or usage patterns out there?

EDIT: Just noticed that https://github.com/swade1987/deprek8ion/issues/18 is quite related to what I'm saying. I think I would prefer the folder structure approach. Then I can use the latest version and choose which K8S version(s?) I'm targeting. I did also in my usage end up just cloning the repo and using the policies manually with conftest to run the specific checks I needed.
